// api/apiService.js
import axios from 'axios'

const baseUrl = 'https://conduit.productionready.io/api'

const prepareUrl = url => baseUrl + url;

class ApiService {
    constructor() {
        this.service = axios.create()
        this.service.interceptors.request.use((config) => {
            if (localStorage.getItem('token')) {
                config.headers.authorization = 'Token ' + localStorage.getItem('token')
            }
            return config
        })
        this.service.interceptors.response.use(
            response => response.data, 
            (error) => {
                // Network Error ( Offline )
                if ( error.message === 'Network Error') {
                    return alert(error.message)
                }
                return Promise.reject(error)
            })
    }

    get(url, params = {}) {
        return this.service.get(prepareUrl(url), { params })
    }

    post(url, data = {},  params = {}) {
        return this.service.post(prepareUrl(url), data, { params })
    }

    put(url, data = {},  params = {}) {
        return this.service.put(prepareUrl(url), data, { params })
    }

    patch(url, data = {},  params = {}) {
        return this.service.patch(prepareUrl(url), data, { params })
    }

    delete(url, params = {}) {
        return this.service.delete(prepareUrl(url), { params })
    }
}

export default new ApiService();
