// api/apiService.js
import axios from 'axios'

const baseUrl = 'https://conduit.productionready.io/api'

const prepareUrl = url => baseUrl + url;

class ApiService {
    constructor() {
        this.service = axios.create()
        this.service.interceptors.request.use((config) => {
            // Adds headers for request
            console.log('Request Config', config)
            return config
        }, (error) => {
            console.log('Request error', )
            return Promise.reject(error)
        })
        this.service.interceptors.response.use(
            response => response.data, 
            (error) => {
                // if (error.response.status === 401 || error.response.status === 403) {
                //     document.location.assign('/login');
                // }
                return Promise.reject(error)
            })
    }

    get(url, params = {}) {
        return this.service.get(prepareUrl(url), { params })
    }

    post(url, data = {},  params = {}) {
        return this.service.post(prepareUrl(url), data, { params })
    }

    put(url, data = {},  params = {}) {
        return this.service.put(prepareUrl(url), data, { params })
    }

    patch(url, data = {},  params = {}) {
        return this.service.patch(prepareUrl(url), data, { params })
    }

    delete(url, params = {}) {
        return this.service.delete(prepareUrl(url), { params })
    }
}

export default new ApiService();
